contracts/ExponentialReverseDutchAuction.sol

```diff
@@ -0,0 +1,199 @@
+        if(x >= 61 * 1e3) {
+            return 0;
+        }else{
+            uint256 scaledPower=x/1e3;
+            uint256 remainder=x%1e3;
+            
+            if(remainder==0) return decayLookup[scaledPower];
+
+            uint256 higherValue=decayLookup[scaledPower];
+            uint256 lowerValue=scaledPower<61? decayLookup[scaledPower+1] : 0;
+            return higherValue - ((higherValue-lowerValue)*remainder)/1e3;  //Linear interpolation
+        }
+    }
+
+    function getCurrentPrice(uint256 auctionId) public view validAuctionId(auctionId) returns (uint256) {
+        AuctionData storage auction=auctions[auctionId];
+        require(block.timestamp<auction.deadline,"Auction has ended");
+        require(!auction.isClaimed,"Auction has ended");
+
+        uint256 timeElapsed=block.timestamp-(auction.deadline-auction.duration);
+        uint256 x=timeElapsed * auction.decayFactor;
+        uint256 decayValue=getDecayValue(x);
+
+        // price(t) = reservedPrice + (startingPrice - reservedPrice) * 2^(-timeElapsed*decayFactor)
+        return auction.reservedPrice + ((auction.startingPrice-auction.reservedPrice)*decayValue)/1e18; 
+    }
+
+    //placeBid function is not required in this auction type as the price is determined by the auctioneer and the auctioneer can withdraw the item on placing the bid only
+    function withdrawItem(uint256 auctionId,uint256 bidAmount) external validAuctionId(auctionId) {
+        AuctionData storage auction = auctions[auctionId];
+        require(block.timestamp<auction.deadline,"Auction has ended");
+        require(!auction.isClaimed,"Auction has been settled");
+
+        uint256 currentPrice=getCurrentPrice(auctionId);
+        require(bidAmount>=currentPrice,"Bid amount is less than current price");
+        
+        SafeERC20.safeTransferFrom(IERC20(auction.biddingToken),msg.sender,address(this),currentPrice);
+        if(auction.auctionType == AuctionType.NFT){
+            IERC721(auction.auctionedToken).safeTransferFrom(address(this),msg.sender,auction.auctionedTokenIdOrAmount);
+        }else{
+            SafeERC20.safeTransfer(IERC20(auction.auctionedToken),msg.sender,auction.auctionedTokenIdOrAmount);
+        }
+
+        auction.winner=msg.sender;
+        auction.availableFunds=currentPrice;
+        auction.isClaimed=true;
+
+        emit itemWithdrawn(auctionId,msg.sender,auction.auctionedToken,currentPrice);
+    }
+
+    function withdrawFunds(uint256 auctionId) external validAuctionId(auctionId){
+        AuctionData storage auction = auctions[auctionId]; 
+        require(msg.sender==auctions[auctionId].auctioneer,"Not auctioneer!");
+        uint256 withdrawAmount=auction.availableFunds;
+        require(withdrawAmount > 0,"No funds available");
+
+        SafeERC20.safeTransfer(IERC20(auction.biddingToken),auction.auctioneer,withdrawAmount);
+        auction.availableFunds=0;
+
+        emit fundsWithdrawn(
+            auctionId,
```

Comment:
There is no need to break lines here.  Look for other places where the formatter is exaggerating in the line breaks and eliminate the unnecessary line breaks.